CONTRIBUTING.md

```diff
@@ -52,31 +52,66 @@ It is unnecessary to manually edit (rebase) your commit history. If, however, yo
 
 ### Building and testing locally
 
-As described in [the README](https://github.com/scikit-hep/awkward-1.0#installation-for-developers), you can build locally using
+As described in [the README](readme.md), Awkward Array is shipped as two packages: `awkward` and `awkward-cpp`. The `awkward-cpp` package contains the compiled C++ components required for performance. If you do not need to frequently modify this code, then it can simply be installed using `pip`.
 
+Subsequent steps require the generation of code and datafiles (kernel specification, header-only includes). This can be done with the `prepare` nox session:
 ```bash
-python localbuild.py --pytest tests
+nox -s prepare
 ```
 
-The `--pytest tests` runs the integration tests from the `tests` directory (drop it to build only).
+The `prepare` session accepts flags to specify exact generation targets, e.g.
+```bash
+nox -s prepare -- --tests --docs
+```
+This can reduce the time taken to perform the preparation step in the event that only the package-building step is needed.
+
+#### Installing the `awkward-cpp` package
+
+The C++ components can be installed by building the `awkward-cpp` package:
+```bash
+python -m pip install ./awkward-cpp
+```
+
+If you are working on the C++ components of Awkward Array, it might be more convenient to skip the build isolation step, which involves creating an isolated build environment. First, you must install the build requirements:
+```bash
+python -m pip install scikit-build-core pybind11 ninja cmake
+```
+
+Then the installation can be performed without build isolation:
+```bash
+python -m pip install --no-build-isolation ./awkward-cpp
+```
```

Comment:
It's because `pyproject_metadata`, `distlib`, and `pathspec` are also dependencies of scikit-build-core (but only when building wheels or sdists).

Reply:
(As well needing to install the git version until I release, that too)

Reply:
`build` does, you are not using `build`, you are using `pip`. For `pip` it's an extra flag because adding it by default is not backward compatible: `python -m pip install --no-build-isolation --check-build-dependencies ./awkward-cpp`

Reply:
(These are pretty light weight dependencies so I might add them by default at some point, especially for things like this. Though once scikit-build uses scikit-build-core it will likely include everything needed)

Reply:
Just released the next version: https://github.com/scikit-build/scikit-build-core/releases/tag/v0.1.0a1 on PyPI.

You can do `scikit-build-core[pyproject]` to get the sdist/wheel dependencies directly. Might be a good idea for the requires, too.